ktor-client/ktor-client-okhttp/jvm/src/io/ktor/client/engine/okhttp/OkUtils.kt

```diff
@@ -4,19 +4,35 @@
 
 package io.ktor.client.engine.okhttp
 
+import io.ktor.client.features.*
 import io.ktor.http.*
 import kotlinx.coroutines.*
 import okhttp3.*
 import okhttp3.Headers
 import java.io.*
+import java.net.*
 import kotlin.coroutines.*
 
 internal suspend fun OkHttpClient.execute(request: Request): Response = suspendCancellableCoroutine {
     val call = newCall(request)
     val callback = object : Callback {
 
         override fun onFailure(call: Call, cause: IOException) {
-            if (!call.isCanceled) it.resumeWithException(cause)
+            if (call.isCanceled) {
+                return
```

Comment:
Don't we get stuck because of a never resumed continuation?

Reply:
Well, again, I haven't changed this behavior, the current code looks the following way: https://github.com/ktorio/ktor/blob/master/ktor-client/ktor-client-okhttp/jvm/src/io/ktor/client/engine/okhttp/OkUtils.kt#L18-L20

If the call is not canceled I guess `onResponse` will be called or the coroutine will be canceled eventually.

Reply:
Ah, sorry, didn't notice the negation in the old version